polyfill/lib/date.mjs

```diff
@@ -0,0 +1,286 @@
+  const {
+    year = this[SLOT_YEAR],
+    month = this[SLOT_MONTH],
+    day = this[SLOT_DAY]
+  } = dateTimeLike;
+  return new Date(year, month, day, disambiguation);
+};
+Date.prototype.plus = function plus(
+  durationLike = {},
+  disambiguation = "constrain"
+) {
+  const duration = ES.CastToDuration(durationLike);
+  let { year, month, day } = this;
+  let {
+    years,
+    months,
+    days,
+    hours,
+    minutes,
+    seconds,
+    milliseconds,
+    microseconds,
+    nanoseconds
+  } = duration;
+  if (
+    hours ||
+    minutes ||
+    seconds ||
+    milliseconds ||
+    microseconds ||
+    nanoseconds
+  )
+    throw new RangeError("invalid duration");
+  ({ year, month, day } = ES.AddDate(
+    year,
+    month,
+    day,
+    years,
+    months,
+    days,
+    disambiguation
+  ));
+  ({ year, month, day } = ES.BalanceDate(year, month, day));
+  return new Date(year, month, day);
+};
+Date.prototype.minus = function minus(
+  durationLike = {},
+  disambiguation = "constrain"
+) {
+  const duration = ES.CastToDuration(durationLike);
+  let { year, month, day } = this;
+  let {
+    years,
+    months,
+    days,
+    hours,
+    minutes,
+    seconds,
+    milliseconds,
+    microseconds,
+    nanoseconds
+  } = duration;
+  if (
+    hours ||
+    minutes ||
+    seconds ||
+    milliseconds ||
+    microseconds ||
+    nanoseconds
+  )
+    throw new RangeError("invalid duration");
+  ({ year, month, day } = ES.SubtractDate(
+    year,
+    month,
+    day,
+    years,
+    months,
+    days,
+    disambiguation
+  ));
+  ({ year, month, day } = ES.BalanceDate(year, month, day));
+  return new Date(year, month, day);
+};
+Date.prototype.difference = function difference(
+  other,
+  disambiguation = "constrain"
+) {
+  const [one, two] = [this, other].sort(DateTime.compare);
+  let years = two.year - one.year;
+
+  let days =
+    ES.DayOfYear(two.year, two.month, two.day) -
+    ES.DayOfYear(one.year, one.month, one.day);
+  if (days < 0) {
+    years -= 1;
+    days = (ES.LeapYear(two.year) ? 366 : 365) + days;
+  }
+  if (
+    disambiguation === "constrain" &&
+    month === 2 &&
+    ES.LeapYear(one.year) &&
+    !ES.LeapYear(one.year + years)
+  )
+    days + 1;
+
+  if (days < 0) {
+    years -= 1;
+    days += ES.DaysInMonth(two.year, two.month);
+  }
+  const Duration = ES.GetIntrinsic("%Temporal.Duration%");
+  return new Duration(years, 0, days, 0, 0, 0, 0, 0, 0);
+};
+Date.prototype.toString = function toString() {
+  let year = ES.ISOYearString(this[SLOT_YEAR]);
+  let month = ES.ISODateTimePartString(this[SLOT_MONTH]);
+  let day = ES.ISODateTimePartString(this[SLOT_DAY]);
+  let resultString = `${year}-${month}-${day}`;
+  return resultString;
+};
+Date.prototype.toLocaleString = function toLocaleString(...args) {
+  return new Intl.DateTimeFormat(...args).format(this);
+};
+Date.prototype.toJSON = function toJSON() {
+  return this.toString();
```

Comment:
should this observably call toString on the receiver, or should it just === Date.prototype.toString?